swim/state_transitions.go

```diff
@@ -46,30 +47,40 @@ type transitionTimer struct {
 type stateTransitions struct {
 	sync.Mutex
 
-	node *Node
-
+	node     *Node
+	clock    clock.Clock
```

Comment:
Have we considered the edge case where one node could end up in a different spacetime continuum? Then we might need multiple clocks to track the different continuums (and keep the cluster stable).

This is a very real concern, considering we are all mocking time in our Amsterdam office and the Large Hadron Collider is only roughly 400 miles away. Just saying.